### PR TITLE
WIP: add support to add/remove reactions

### PR DESCRIPTION
**Pull request by arpith**
This PR adds support to add or remove an emoji reaction to a message. 
The usage is `zulip.reactions.add({ message_id: 1, emoji: smile })` and `zulip.reactions.remove({ message_id: 1, emoji: smile })`

This is a work in progress, and review/feedback would be appreciated!

### CODE REVIEW

File: src/resources/reactions.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const api = require('../api');
+
+function reactions(config) {
+  const url = `${config.apiURL}/reactions`;
+  const call = (method, params) => api(url, config, method, params);
+  return {
+    add: params => call('POST', params),
```
Comment (aero31aero) [inline src/resources/reactions.js]: We should also have a method for GETting the existing reactions on a message, perhaps.

**Comment by arpith** [comment]
Ah I forgot to unmark it as WIP!

**Comment by aero31aero** [comment]
Merged, thanks @arpith! About getting message reactions alone, I suppose not.

**Comment by timabbott** [comment]
For this one, we should be using the new reactions API:
```
    # New endpoint for handling reactions.                                                           
    url(r'^messages/(?P<message_id>[0-9]+)/reactions$',                                              
        rest_dispatch,                                                                               
        {'POST': 'zerver.views.reactions.add_reaction',                                              
         'DELETE': 'zerver.views.reactions.remove_reaction'}),                                       
```

It seems like you're doing a POST, but I don't understand why you're passing the message ID with a URL parameter?

**Pull request merged**